File: app/src/main/java/com/jinjoo/pouch/adapter/SearchAdapter.java
package com.jinjoo.pouch.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.jinjoo.pouch.R;
import com.jinjoo.pouch.model.Cosmetic;

import java.util.List;

public class SearchAdapter extends RecyclerView.Adapter<SearchAdapter.ViewHolder> {

    Context context;

    List<Cosmetic> items;


    //어댑터 생성자 -> getApplicationContext, 리스트로 보여줄 아이템
    public SearchAdapter(Context context, List<Cosmetic> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(context).inflate(R.layout.item_search, parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        Glide.with(context).load(items.get(position).getImg()).into(holder.imageView);
        holder.brand.setText(items.get(position).getBrand());
        holder.name.setText(items.get(position).getName());

        holder.regi.setOnClickListener(view -> {
            //Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(imgData.getLink()));
            //startActivity(context,intent,null);
        });


    }

    @Override
    public int getItemCount() {
        return items.size();
    }





    //뷰홀더
    class ViewHolder extends RecyclerView.ViewHolder{
        ImageView imageView;
        TextView brand;
        TextView name;
        Button regi;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.img);
            brand = itemView.findViewById(R.id.brand);
            name = itemView.findViewById(R.id.name);
            regi = itemView.findViewById(R.id.regi);
        }
    }
}

File: app/src/main/kotlin/com/jinjoo/pouch/adapter/SearchAdapter.kt
package com.jinjoo.pouch.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.jinjoo.pouch.GlideApp
import com.jinjoo.pouch.R
import com.jinjoo.pouch.api.model.Cosmetic
import kotlinx.android.synthetic.main.item_search.view.*

class SearchAdapter : RecyclerView.Adapter<SearchAdapter.ViewHolder>() {

    private var items: MutableList<Cosmetic> = mutableListOf()
    private var listener: ItemClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(parent)

    fun setItemClickListener(listener: ItemClickListener?) {
        this.listener = listener
    }

    fun setItems(items: MutableList<Cosmetic>) {
        this.items = items
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        items[position].let { item ->
            with(holder.itemView) {
                GlideApp.with(context).load(items[position].img).into(img)
                brand.text = item.brand
                name.text = item.name
                regi.setOnClickListener { listener?.onItemClick(item) }
            }

        }
    }

    override fun getItemCount(): Int {
        return items.size
    }

    //뷰홀더
    class ViewHolder(parent: ViewGroup) : RecyclerView.ViewHolder(
            LayoutInflater.from(parent.context)
                    .inflate(R.layout.item_search, parent, false)
    )

    interface ItemClickListener {

        fun onItemClick(cos: Cosmetic)
    }
}